Queueomatic/Server/Endpoints/User/Delete/DeleteUserEndpoint.cs

```diff
@@ -1,27 +1,33 @@
 ﻿using FastEndpoints;
+using Queueomatic.DataAccess.UnitOfWork;
+using Queueomatic.Server.Services.AuthenticationService;
 
 namespace Queueomatic.Server.Endpoints.User.Delete;
 
-public class DeleteUserEndpoint: Endpoint<DeleteUserRequest, DeleteUserResponse>
+public class DeleteUserEndpoint: Endpoint<DeleteUserRequest>
 {
+    private readonly IUnitOfWork _unitOfWork;
+
+    public DeleteUserEndpoint(IUnitOfWork unitOfWork)
+    {
+        _unitOfWork = unitOfWork;
+    }
     public override void Configure()
     {
-        Verbs(Http.DELETE);
-        Routes("/users/{email}");
-        AllowAnonymous();
+        Delete("/users/{email}");
+        Policies("SignedInUser");
     }
 
     public override async Task HandleAsync(DeleteUserRequest req, CancellationToken ct)
     {
-        try
-        {
-            var response = new DeleteUserResponse();
-            await SendAsync(response, 204, cancellation: ct);
-        }
-        catch (TaskCanceledException exception)
-            when(exception.CancellationToken == ct)
+        await  _unitOfWork.UserRepository.DeleteAsync(req.Email);
+        var result = await _unitOfWork.SaveAsync();
+        if (result == 0)
         {
-            Logger.LogInformation($"Task {nameof(DeleteUserEndpoint)} was cancelled.");
+            await SendNotFoundAsync();
+            return;
         }
+        await SendOkAsync("Account deleted");
```

Comment:
I suggest we send a 204 response code as it is in the API specification. There is a send method from FastEndpoints called `SendNoContentAsync()`.